### ISSUE TITLE
How to enable LSR as described in Attentional Encoder Network for Targeted Sentiment Classification?

### ISSUE THREAD

**Issue by fhamborg** [songyouwei/ABSA-PyTorch#93]
In the paper Attentional Encoder Network for Targeted Sentiment Classification, you show that AEN with LSR yields a significant improvement over AEN without LSR. However, when looking in the code, the respective class https://github.com/songyouwei/ABSA-PyTorch/blob/bf55517ed5cb8dda5960b15575e87dd06101586d/models/aen.py#L16 is actually never being used. How to use it? Why did you disable it?

**Comment by songyouwei** [comment]
LSR is not used by default because other implemented models do not use this trick.

https://github.com/songyouwei/ABSA-PyTorch/blob/bf55517ed5cb8dda5960b15575e87dd06101586d/train.py#L23

You can enable it by replacing `nn.CrossEntropyLoss()` with `CrossEntropyLoss_LSR()`

https://github.com/songyouwei/ABSA-PyTorch/blob/bf55517ed5cb8dda5960b15575e87dd06101586d/train.py#L160

**Comment by fhamborg** [comment]
Alright, that's awesome. Thank you very much =)

**Comment by fhamborg** [comment]
Do you happen to know how to weight the loss of different classes differently? E.g., our dataset is pretty imbalanced and we'd like to weight classes accordingly. In the regular `CrossEntropyLoss` provided by pytorch, there is a [weight parameter](https://pytorch.org/docs/stable/nn.html#crossentropyloss) that allows to do that. My current implementation changes your `_toOneHot_smooth` as follows:

```
    def _toOneHot_smooth(self, label, batchsize, classes):
        base_prob = self.para_LSR * 1.0 / classes
        one_hot_label = torch.zeros(batchsize, classes) + base_prob

        if self.weight is not None:
            one_hot_label = one_hot_label * self.weight

        for i in range(batchsize):
            index = label[i]
            one_hot_label[i, index] += (1.0 - self.para_LSR)

        return one_hot_label
```

whereby self.weight is a tensor consisting of C numbers (where C is the number of classes). Does this seem correct to you? I'd very much appreciate your feedback!
